### PR TITLE
Tools: Add analysis file for optical flow calibration data

### PR DESCRIPTION
**Pull request by priseborough**
Calculate and display recommended values of FLOW_FXSCALE and FLOW_FYSCALE from an optical flow sensor calibration test
Save plots to flow_calibration.pdf file
Save calibration parameters to flow_calibration.param file

Example plot output:

[flow_calibration.pdf](https://github.com/ArduPilot/ardupilot/files/756777/flow_calibration.pdf)

Example console output:

screen shot 2017-02-07 at 5 51 56 pm

### CODE REVIEW

**Comment by rmackay9** [comment]
ok thanks for this.  It's perhaps a bit odd that it's run as part of the dataflash log analysis stuff.  I thought this was mainly meant to automatically find issues during flight.  Not sure that it was meant for calibrating a sensor.

It might be better to have it as a stand-alone script that gets run on a log file.  In either case, I very much like the feature and if this is the way to ease setup then I'm ok with it.  Thanks!

File: Tools/LogAnalyzer/tests/TestOptFlow.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+            for i in range(len(Roll)):
+                if abs(Roll[i]) > tilt_threshold:
+                    startTime = att_time_us[i]
+                    break
+            for i in range(len(flow_time_us)):
+                if flow_time_us[i] > startTime:
+                    startRollIndex = i
+                    break
+
+            # calculate the end time for the roll calibration
```
Comment (peterbarker) [inline Tools/LogAnalyzer/tests/TestOptFlow.py]: pitch, not roll

**Comment by peterbarker** [comment]
This patch certainly sets a new record for degree of analysis and fanciness of output!

I have only one concern, and that is how the analyser reacts if the log doesn't contain the required information; it doesn't seem to handle the case where the log doesn't contain data to sensibly set the endpoints for the roll and pitch data.

File: Tools/LogAnalyzer/tests/TestOptFlow.py [hunk 2]
```diff
@@ -113,7 +114,7 @@ def WARN():
                     startRollIndex = i
                     break
 
-            # calculate the end time for the roll calibration
+            # calculate the end time for the pitch calibration
```
Comment (peterbarker) [inline Tools/LogAnalyzer/tests/TestOptFlow.py]: Sorry Paul, I gave you a bum steer here.  I deleted the comment seconds after I created it, but you picked it up from email, I assume.

I think this *is* roll, isn't it?

**Comment by peterbarker** [comment]
Merged!

**Pull request closed without merging**